### ISSUE TITLE
Remove **kwargs if not used

### ISSUE THREAD

**Issue by sirex** [schematics/schematics#402]
I had fllowing situation:

```python
from schematics.models import Model
from schematics.types import IntType

class Mapping(Model):
    value = IntType(deserialized_from='@value')

assert Mapping({'@value': '1'}, strict=False).to_primitive() == {'value': 1}
# => {'value': None} != {'value': 1}
```

Here I misspelled `deserialized_from` argument name and it took quite some time to figure it out, because I did not suspected, that `IntType` can swallow arguments with any name.

This happened, because `BaseType` accepts `**kwargs` without using it, which allows all type classes to accept any arguments.

If not used, `**kwargs` should be removed from `BaseType` constructor letting Python do automatic passed argument check.

After removing `**kwargs` I will get nice Python error instead of silence:

```pytb
TypeError: __init__() got an unexpected keyword argument 'deserialized_from'
```

**Comment by bintoro** [comment]
Agreed. I have no idea why `BaseType` takes variadic keyword arguments.

**Comment by bintoro** [comment]
Fixed (4b29b16).
